truss/templates/trtllm/packages/build_engine_utils.py

```diff
@@ -0,0 +1,129 @@
+    max_batch_size: Optional[int] = None
+    tp_size: Optional[int] = None
+    pp_size: Optional[int] = None
+    world_size: Optional[int] = None
+    gather_all_token_logits: Optional[bool] = None
+    multi_block_mode: Optional[bool] = None
+    remove_input_padding: Optional[bool] = None
+    use_gpt_attention_plugin: Optional[str] = None
+    paged_kv_cache: Optional[bool] = None
+    use_inflight_batching: Optional[bool] = None
+    enable_context_fmha: Optional[bool] = None
+    use_gemm_plugin: Optional[str] = None
+    use_weight_only: Optional[bool] = None
+    output_dir: Optional[str] = None
+    model_dir: Optional[str] = None
+    ft_model_dir: Optional[str] = None
+    dtype: Optional[str] = None
+    int8_kv_cache: Optional[bool] = None
+    use_smooth_quant: Optional[bool] = None
+    per_token: Optional[bool] = None
+    per_channel: Optional[bool] = None
+    parallel_build: Optional[bool] = None
+
+    # to disable warning because `model_dir` starts with `model_` prefix
+    model_config = ConfigDict(protected_namespaces=())
+
+    def as_command_arguments(self):
+        non_bool_args = [
+            element
+            for arg, value in self.dict().items()
+            for element in [f"--{arg}", str(value)]
+            if value is not None and not isinstance(value, bool)
+        ]
+        bool_args = [
+            f"--{arg}"
+            for arg, value in self.dict().items()
+            if isinstance(value, bool) and value
+        ]
+        return non_bool_args + bool_args
+
+
+class CalibrationConfig(BaseModel):
+    kv_cache: Optional[bool] = None  # either to calibrate kv cache
+    sq_alpha: Optional[float] = None
+
+    def cache_path(self) -> Path:
+        if self.kv_cache is not None:
+            return Path("kv_cache")
+        else:
+            return Path(f"sq_{self.sq_alpha}")
+
+
+class EngineBuildArgs(BaseModel, use_enum_values=True):
+    repo: Optional[str] = None
+    args: Optional[ArgsConfig] = None
+    quant: Optional[Quant] = None
+    calibration: Optional[CalibrationConfig] = None
+    engine_type: Optional[EngineType] = None
+
+    @classmethod
+    def from_config(cls, config: dict):
+        return cls(
+            repo=config["repo"],
+            args=ArgsConfig(**config["args"]),
+            quant=Quant(config["quant"]) if "quant" in config else None,
+            calibration=CalibrationConfig(**config["calibration"])
+            if "calibration" in config
+            else None,
+            engine_type=EngineType(config["engine_type"])
+            if "engine_type" in config
+            else None,
+        )
+
+
+def build_engine(
+    engine_build_args: EngineBuildArgs,
+    dst: Path,
+):
+    import sys
+
+    sys.path.append("/app/baseten")
+
+    import os
+    import shutil
+
+    from build_engine import Engine, build_engine
+    from trtllm_utils import docker_tag_aware_file_cache
+
+    engine = Engine(**engine_build_args.model_dump())
```

Comment:
Is `Engine` an actual engine or just config - does the instantiation of this class already execute the building process?

Reply:
`Engine` is a config, we use that to then kick off the building of an actual Engine.

Reply:
Let's make a follow up ticket to clean up this class name to something less ambiguous so we know this is the configuration used to build an engine from